### ISSUE TITLE
Scaffold-DbContext can name namespace wrong if phrase with space is used on OutputDir

### ISSUE THREAD

**Issue by hahyes** [dotnet/efcore#24607]
Scaffold-DbContext is naming namespace wrong if `-OutputDir` is filled with phrase containing space.

For example: `Example Folder` OutputDir makes namespace `ConsoleApp.Example Folder`, what is wrong for naming namespaces and can't be builded. It should be `ConsoleApp.Example_Folder`.

### Code

I made command through PMC like this:

```
Scaffold-DbContext "connectionString" MySql.EntityFrameworkCore -Tables some_tables_here -OutputDir "Example Folder" -Force
```

Command is working, but output classes and context have wrong namespace.
### Include provider and version information

EF Core version: 5.0
Database provider: MySql.EntityFrameworkCore
Target framework: NET 5.0
Operating system:
IDE: Visual Studio 2019 16.9.3

**Comment by Psypher9** [comment]
Is this open for contributions?

**Comment by Psypher9** [comment]
@bricelam , so I've been looking at this a little bit and it looks like the private function that gets the subnamespace inside of the `EFCore.Design.DatabaseOperations.ScaffoldContext` needs to have a `.Replace(' ', '_')` added to [this ternary](https://github.com/dotnet/efcore/blob/bb46c5ef74b24d8f17fa98138245603a87ce6a64/src/EFCore.Design/Design/Internal/DatabaseOperations.cs#L149-L154) before the `string.Join('.',...)` is evaluated.

I also happened to notice that the [GetNamespaceFromOutputPath method here](https://github.com/dotnet/efcore/blob/bb46c5ef74b24d8f17fa98138245603a87ce6a64/src/EFCore.Design/Design/Internal/DatabaseOperations.cs#L127) and [SubnamespaceFromOutputPath method here](https://github.com/dotnet/efcore/blob/bb46c5ef74b24d8f17fa98138245603a87ce6a64/src/EFCore.Design/Design/Internal/DatabaseOperations.cs#L142) are the exact same inside of [DbContextOperations.Optimize](https://github.com/dotnet/efcore/blob/bb46c5ef74b24d8f17fa98138245603a87ce6a64/src/EFCore.Design/Design/Internal/DbContextOperations.cs#L142). 

I've been playing around with this in my own branch and I propose making a service to handle generating namespaces based on file paths and 

1.  have both `DbContextOperations.Optimize` and `DatabaseOperations.ScaffoldContext` use that service 
2. have that service use the `.Replace(' ', '_')` to handle spaces in output paths.

Another reason I propose making this a service is that, when I was looking to add tests there wasn't really a good way to make tests for this particular thing for either use case (at least to my knowledge). I've added tests to support my work so far, but is this something you guys really want and if so, do you really want to replaces spaces with `_` or just remove them entirely? 

Thanks!

**Comment by bricelam** [comment]
Yes, I'm OK with this approach. It would also be nice to do it for Migration class names.
